src/main/kotlin/org/rust/lang/core/psi/RustElementFactory.kt

```diff
@@ -42,6 +42,17 @@ object RustElementFactory {
     fun createIdentifier(project: Project, name: String): PsiElement =
         createFromText<RustModDeclItemElement>(project, "mod $name;")!!.identifier
 
+    fun createWhereClause(project: Project, lifetimeBounds: List<RustLifetimeParamElement>, typeBounds: List<RustTypeParamElement>): RustWhereClauseElement? {
+        val lifetimes = lifetimeBounds
+            .filter { it.lifetimeParamBounds != null }
+            .mapNotNull { it.node.text }
```

Comment:
I believe you can use a single `.mapNotNull { it.lifetimeParamBouds?.node?.text }` here.